### PR TITLE
create a prometheus exporter on a standard http port

### PR DESCRIPTION
**Pull request by dat2**
This creates a prometheus exporter in process. This was requested in https://github.com/levkk/pgcat/issues/83.

If you add `enable_prometheus_exporter = true` to `pgcat.toml` then you can make a `GET` request to `<pgcat host>:9930/metrics` to get a list of metrics.

Here's an example of the prometheus metrics exporter endpoint.

Reference: https://prometheus.io/docs/instrumenting/writing_exporters/
```
➜  pgcat git:(nickdujay/metric-exports) ✗ http -v :9930/metrics
GET /metrics HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: localhost:9090
User-Agent: HTTPie/3.2.1



HTTP/1.1 200 OK
content-length: 7259
content-type: text/plain; version=0.0.4

# HELP pgcat_sv_idle How many server connections are idle
# TYPE pgcat_sv_idle gauge
pgcat_sv_idle{shard="0",host="127.0.0.1",database="simple_db",role="primary"} 1

# HELP pgcat_avg_recv Average of total_received every 15 seconds
# TYPE pgcat_avg_recv gauge
pgcat_avg_recv{role="primary",shard="0",host="127.0.0.1",database="simple_db"} 3

# HELP pgcat_total_received Number of bytes received from the server
# TYPE pgcat_total_received counter
pgcat_total_received{host="127.0.0.1",shard="0",database="simple_db",role="primary"} 54

# HELP pgcat_avg_wait_time Average of total_wait_time every 15 seconds
# TYPE pgcat_avg_wait_time gauge
pgcat_avg_wait_time{host="127.0.0.1",role="primary",shard="0",database="simple_db"} 1042
#truncated
```

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -95,6 +97,23 @@ async fn main() {
     };
 
     let config = get_config();
+
+    config.show();
```
Comment (dat2) [inline src/main.rs:101]: whoops, forgot to remove debugging

File: Cargo.toml [hunk 2]
```diff
@@ -31,3 +31,5 @@ base64 = "0.13"
 stringprep = "0.1"
 tokio-rustls = "0.23"
 rustls-pemfile = "1"
+hyper = { version = "0.14", features = ["full"] }
+phf = { version = "0.10", features = ["macros"] }
```
Comment (levkk) [inline Cargo.toml:35]: Terminating new line

File: src/config.rs [hunk 3]
```diff
@@ -308,6 +311,12 @@ impl Config {
                 info!("TLS support is disabled");
             }
         };
+        if let Some(true) = self.general.enable_prometheus_exporter {
+            info!(
+                "Exposing prometheus metrics on http://{}:{}/metrics",
```
Comment (levkk) [inline src/config.rs:316]: This will print on every config reload I believe. Could you move that into `main.rs` when it launches the http listener?

**Review by levkk — APPROVED** [review]
Amazing, thank you so much!

**Pull request merged**